common/filetoolkitwithundo.cpp

```diff
@@ -180,15 +123,33 @@ bool FileToolkitWithUndo::overlayMount(const std::string &lower,
     std::string fstype = "overlay";
     unsigned long flags = 0;
 
-    if (!createDirectory(lower)
-        || !createDirectory(upper)
-        || !createDirectory(work)
-        || !createDirectory(dst))
+    if (!m_create.createDirectory(lower)
+        || !m_create.createDirectory(upper)
+        || !m_create.createDirectory(work)
+        || !m_create.createDirectory(dst))
     {
+        m_create.rollBack();
         log_error() << "Failed to create lower/upper/work directory for overlayMount. lowerdir=" <<
                        lower << ", upperdir=" << upper << ", workdir=" << work;
         return false;
     }
+    m_create.clear();
```

Comment:
In the went wrong step, the local ft would go out of scope and would automatically cleanup as part of its dtor.